source/LinkedList.inl

```diff
@@ -94,21 +127,21 @@ DataStructures::LinkedList<T>::~LinkedList()
 
 template<typename T>
 size_t
-DataStructures::LinkedList<T>::size()
+DataStructures::LinkedList<T>::size() const
 {
   return number_of_elements;
 }
 
 template<typename T>
 bool
-DataStructures::LinkedList<T>::empty()
+DataStructures::LinkedList<T>::empty() const
 {
   return number_of_elements == 0;
 }
 
 template<typename T>
 bool
-DataStructures::LinkedList<T>::operator==(LinkedList<T>& other)
+DataStructures::LinkedList<T>::operator==(LinkedList<T>& other) const
```

Comment:
Look in to `std::equal` for the post-size-check implementation here.

Reply:
All these niceties in C++ are _so_ unstable across standards.  No version of std::equal that's valid in C++14 is valid in C++20.